packages/minter-contracts/ligo/src/minter_collection/editions/fa2_multi_nft_token_editions.mligo

```diff
@@ -60,23 +60,30 @@ let mint_editions ( editions_list , storage : mint_edition_run list * editions_s
   let new_storage = List.fold mint_single_edition_run editions_list storage in
   ([] : operation list), new_storage
 
-let distribute_edition_to_addresses ( edition_id,receivers, edition_metadata, storage : nat * (address list) * edition_metadata * editions_storage)
-  : editions_storage * edition_metadata = 
-  let distribute_edition_to_address : ((editions_storage * edition_metadata) * address) -> (editions_storage * edition_metadata) = 
-    fun ( (storage, edition_metadata), to_  : (editions_storage * edition_metadata) * address ) ->
+let distribute_edition_to_addresses ( edition_id, token_info, receivers, edition_metadata, storage : nat *  ((string, bytes) map) * (address list) * edition_metadata * editions_storage)
+  : editions_storage = 
+  let distribute_edition_to_address : ((mint_tokens_param * edition_metadata) * address) -> (mint_tokens_param * edition_metadata) = 
+    fun ( (mint_tokens_param, edition_metadata), to_  : (mint_tokens_param * edition_metadata) * address) ->
       let u : unit = if (edition_metadata.number_of_editions_to_distribute  > 0) then ()
         else (failwith "NO_EDITIONS_TO_DISTRIBUTE" : unit) in 
       let mint_token_param : mint_token_param = {
           owner = to_;
           token_metadata = {
             token_id = edition_metadata.initial_token_id + abs (edition_metadata.number_of_editions - edition_metadata.number_of_editions_to_distribute); 
-            token_info = Map.literal [(("edition_id" : string), (Bytes.pack edition_id))];
+            token_info = token_info;
           }
       } in
-      let _ , nft_token_storage = mint_edition (mint_token_param, storage.nft_asset_storage.assets) in
-      {storage with nft_asset_storage = {storage.nft_asset_storage with assets = nft_token_storage}}, 
-      {edition_metadata with number_of_editions_to_distribute = edition_metadata.number_of_editions_to_distribute - 1} in
-  (List.fold distribute_edition_to_address receivers (storage, edition_metadata))
+      ((mint_token_param :: mint_tokens_param) , 
```

Comment:
Do I understand it correctly, that you intentionally reverse `mint_tokens_param`, so that when it is later passed to `mint_editions_to_addresses`, resulting operations, being reversed again, restore their original order?

Reply:
No, the order of the `mint_token_params` does not matter because we already include the `token_id` in `mint_token_param`